src/App.tsx

```diff
@@ -15,37 +19,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_BY_ALPHABET = 'alphabet',
+  SORT_BY_LENGTH = 'length',
+  NO_SORTING = '',
+}
+
+function prepareGoods(
+  goods: string[],
+  sortBy: SortType,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortBy) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortBy) {
+        case SortType.SORT_BY_ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.SORT_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState<SortType>(SortType.NO_SORTING);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = prepareGoods(goodsFromServer, sortBy, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortBy !== SortType.SORT_BY_ALPHABET,
+          })}
+          onClick={() => setSortBy(SortType.SORT_BY_ALPHABET)}
```

Comment:
Consider toggling the sort type when the same button is clicked again. Currently, clicking the same sort button repeatedly will not toggle the sorting state. You might want to check if the current `sortBy` is the same as the clicked one and set it to `NO_SORTING` if it is.